src/NuGet.Clients/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -199,6 +201,30 @@ public override async Task ExecuteCommandAsync()
             }
         }
 
+        private static string GetSmallestSdkAnalysisLevel(IReadOnlyList<PackageSpec> projects)
+        {
+            string sdkAnalysisLevel = null;
+            foreach (PackageSpec project in projects)
+            {
+                if (sdkAnalysisLevel == null)
+                {
+                    if (!string.IsNullOrEmpty(project.SdkAnalysisLevel))
+                    {
+                        sdkAnalysisLevel = project.SdkAnalysisLevel;
+                    }
+                }
+                else
+                {
+                    if (string.Compare(sdkAnalysisLevel, project.SdkAnalysisLevel, StringComparison.OrdinalIgnoreCase) < 0)
```

Comment:
This is going to have unexpected results if some projects have SdkAnalysisLevel `10.0.100` and others have `9.0.100`.